src/core/tsi/ssl_transport_security_utils.h

```diff
@@ -168,6 +168,13 @@ absl::StatusOr<std::string> AkidFromCertificate(X509* cert);
 // crl.
 // return: a std::string of the DER encoding of the AKID or a status on failure.
 absl::StatusOr<std::string> AkidFromCrl(X509_CRL* crl);
+
+// Returns true if cert_chain_pem contains a non-empty sequence of PEM
+// certificate blocks.
+bool IsPemCertificateChainNonEmptyAndValid(absl::string_view cert_chain_pem);
+
+// Returns true if private_key_pem contains a non-empty PEM private key block.
```

Comment:
This comment does not match the implementation. It talks about whether it contains a private key, but you're specifically checking for RSA and EC keys. (Which still may not be usable for TLS, e.g. if they're not the expected key type, or if they don't match the certificate.)

What exactly is this API meant to be used for?

Reply:
I think this comment is obsolete after switching to use `PEM_read_bio_PrivateKey`.